### PR TITLE
updated google auth to v2 and ability to change response type

### PR DESCRIPTION
**Pull request by k1ng440**

### CODE REVIEW

File: src/google-oauth2.js [hunk 1]
```diff
@@ -75,14 +79,23 @@ GO2.receiveMessage = function GO2_receiveMessage() {
       hash.replace(/^.*state=go2_([^&]+).*$/, '$1')
     );
   }
+
+  if (go2 && window.location.search.indexOf('code=')) {
```
Comment (timdream) [inline src/google-oauth2.js]: `window.location.search.indexOf('code=') != -1`

**Comment by timdream** [comment]
This looks great! Thank you very much. Could you also update the tests too? We should make sure all features don't break because of this change.

**Comment by k1ng440** [comment]
i have updated the test. please merge it

**Comment by timdream** [comment]
Superb!

**Pull request merged**